### PR TITLE
chore(keyvault-v3): key vault plugin v3

### PR DESCRIPTION
**Pull request by jayzhang**

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/keyvault/v3/index.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      },
+      Reference: {
+        m365ClientSecretReference: Constants.KeyVaultBicep.m365ClientSecretReference,
+        botClientSecretReference: Constants.KeyVaultBicep.botClientSecretReference,
+      },
+    };
+    return ok([{ kind: "bicep", template: result }]);
+  }
+  @hooks([CommonErrorHandlerMW({ telemetry: { component: BuiltInFeaturePluginNames.keyVault } })])
+  async addFeature(
```
Comment (formulahendry) [inline packages/fx-core/src/plugins/resource/keyvault/v3/index.ts:73]: What's 'addFeature' used for?

File: packages/fx-core/src/plugins/resource/keyvault/v3/index.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
+    const activeResourcePlugins = solutionSettings.activeResourcePlugins;
+    if (!activeResourcePlugins.includes(this.name)) activeResourcePlugins.push(this.name);
+    const azureResources = solutionSettings.azureResources;
+    if (!azureResources.includes(AzureResourceKeyVault.id))
+      azureResources.push(AzureResourceKeyVault.id);
+    return ok(armRes.value);
+  }
+  @hooks([CommonErrorHandlerMW({ telemetry: { component: BuiltInFeaturePluginNames.keyVault } })])
+  async afterOtherFeaturesAdded(
```
Comment (formulahendry) [inline packages/fx-core/src/plugins/resource/keyvault/v3/index.ts:88]: Is it the same function as 'updateArmTemplates' in v1?

Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/keyvault/v3/index.ts:88]: Yes

File: packages/fx-core/src/plugins/resource/keyvault/v3/index.ts (hunk 1 above)
Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/keyvault/v3/index.ts:73]: addFeature means enable some feature by adding artifacts, which includes anything like scaffold, generateArmTemplate, update manifest

File: packages/fx-core/src/plugins/resource/keyvault/v3/index.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import { Bicep } from "../../../../common/constants";
+import { generateBicepFromFile } from "../../../../common/tools";
+import { CommonErrorHandlerMW } from "../../../../core/middleware/CommonErrorHandlerMW";
+import { getTemplatesFolder } from "../../../../folder";
+import { AzureResourceKeyVault } from "../../../solution/fx-solution/question";
+import { BuiltInFeaturePluginNames } from "../../../solution/fx-solution/v3/constants";
+import { Constants } from "../constants";
+
+@Service(BuiltInFeaturePluginNames.keyVault)
+export class KeyVaultPluginV3 implements v3.FeaturePlugin {
```
Comment (formulahendry) [inline packages/fx-core/src/plugins/resource/keyvault/v3/index.ts:27]: Just to confirm: No 'activate' function in v3, right?

Comment (jayzhang) [reply packages/fx-core/src/plugins/resource/keyvault/v3/index.ts:27]: Yes, all plugins are added by addFeature command

**Review by formulahendry — CHANGES_REQUESTED** [review]
Please fix the UT.

**Review by formulahendry — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Pull request merged**